This is a shared_module so it gets no SONAME, but it is
linked to, which results in some awkward behaviors, notably
the NEEDED becoming a path like src/core/devices/wwan/libnm-wwan.so

--- a/src/core/devices/wwan/meson.build
+++ b/src/core/devices/wwan/meson.build
@@ -17,7 +17,7 @@ libnm_wwan = shared_module(
     libsystemd_dep,
     mm_glib_dep,
   ],
-  link_args: '-Wl,--version-script,@0@'.format(linker_script),
+  link_args: ['-Wl,--version-script,@0@'.format(linker_script), '-Wl,-soname,libnm-wwan.so'],
   link_depends: linker_script,
   install: true,
   install_dir: nm_plugindir,
